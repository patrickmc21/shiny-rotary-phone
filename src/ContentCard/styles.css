.content-card {
  display: inline-block;
  border: 1px solid chartreuse;
  margin: 4px;
  height: 150px;
  width: 10vw;
}