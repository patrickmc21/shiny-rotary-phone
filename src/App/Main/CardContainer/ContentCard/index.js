import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const ContentCard = () => {
  return(
    <article className='content-card'>
      I'm the ContentCard
    </article>
  )
};

export default ContentCard;