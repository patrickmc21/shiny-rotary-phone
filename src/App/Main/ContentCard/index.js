import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const ContentCard = () => {
  return(
    <div>I'm the content card</div>
  )
}

export default ContentCard;