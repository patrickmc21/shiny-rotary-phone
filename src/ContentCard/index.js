import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const ContentCard = ({categoryItem, activeCategoryName, addToFavorites, buttonClass}) => {
  const catKeys = Object.keys(categoryItem);
  const cardContent = catKeys.filter(key => key !== 'name').map(item => {
    return <li key={item}>{item}: {categoryItem[item]}</li>
  })

  return(
    <article className={`content-card ${activeCategoryName} ${buttonClass}`}>
      <header>
        <button 
          className={`favorite-btn ${buttonClass}`} 
          onClick={() => addToFavorites(categoryItem)}></button>
        <h3>{categoryItem.name}</h3>
      </header>
      <ul>
        {cardContent}
      </ul>
    </article>
  )
};

ContentCard.propTypes = {
  categoryItem: PropTypes.object,
  activeCategoryName: PropTypes.string,
  addToFavorites: PropTypes.func,
  buttonClass: PropTypes.string
}

export default ContentCard;