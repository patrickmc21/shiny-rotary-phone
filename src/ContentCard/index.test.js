import React from 'react';
import { shallow } from 'enzyme';
import ContentCard from './index.js';

describe('ContentCard', () => {
  let mockCategoryItem;
  let mockAddToFavorites;
  let mockButtonClass;
  let mockCategoryName;
  let wrapper;

  beforeEach( () => {
    mockCategoryItem = {
      name: 'Chewbacca',
      Homeworld: 'Kashyyk',
      'Homeworld Population': 1000010,
      Species: 'Wookie'
    };
    mockCategoryName = 'people';
    mockAddToFavorites = jest.fn();
    mockButtonClass = '';
    wrapper = shallow(
      <ContentCard 
        categoryItem={mockCategoryItem}
        activeCategoryName={mockCategoryName}
        addToFavorites={mockAddToFavorites}
        buttonClass={mockButtonClass}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should run addToFavorites on click with correct params', () => {
    wrapper.find('.favorite-btn').simulate('click');
    expect(mockAddToFavorites).toHaveBeenCalledWith(mockCategoryItem);
  });
});