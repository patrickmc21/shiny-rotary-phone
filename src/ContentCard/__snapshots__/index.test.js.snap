// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContentCard
    activeCategoryName="people"
    addToFavorites={[Function]}
    buttonClass=""
    categoryItem={
        Object {
            "Homeworld": "Kashyyk",
            "Homeworld Population": 1000010,
            "Species": "Wookie",
            "name": "Chewbacca",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h3>
                    Chewbacca
          </h3>
          <button
                    className="favorite-btn "
                    onClick={[Function]}
          >
                    Favorite
          </button>
</header>,
        <ul>
          <li>
                    Homeworld
                    : 
                    Kashyyk
          </li>
          <li>
                    Homeworld Population
                    : 
                    1000010
          </li>
          <li>
                    Species
                    : 
                    Wookie
          </li>
</ul>,
      ],
      "className": "content-card  people",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Chewbacca
</h3>,
            <button
              className="favorite-btn "
              onClick={[Function]}
>
              Favorite
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Chewbacca",
            },
            "ref": null,
            "rendered": "Chewbacca",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorite",
              "className": "favorite-btn ",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Favorite",
            "type": "button",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              Homeworld
              : 
              Kashyyk
</li>,
            <li>
              Homeworld Population
              : 
              1000010
</li>,
            <li>
              Species
              : 
              Wookie
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Homeworld",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld",
                ": ",
                "Kashyyk",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Homeworld",
              ": ",
              "Kashyyk",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "Homeworld Population",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld Population",
                ": ",
                1000010,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Homeworld Population",
              ": ",
              1000010,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "Species",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Species",
                ": ",
                "Wookie",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Species",
              ": ",
              "Wookie",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h3>
                        Chewbacca
            </h3>
            <button
                        className="favorite-btn "
                        onClick={[Function]}
            >
                        Favorite
            </button>
</header>,
          <ul>
            <li>
                        Homeworld
                        : 
                        Kashyyk
            </li>
            <li>
                        Homeworld Population
                        : 
                        1000010
            </li>
            <li>
                        Species
                        : 
                        Wookie
            </li>
</ul>,
        ],
        "className": "content-card  people",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Chewbacca
</h3>,
              <button
                className="favorite-btn "
                onClick={[Function]}
>
                Favorite
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Chewbacca",
              },
              "ref": null,
              "rendered": "Chewbacca",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorite",
                "className": "favorite-btn ",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Favorite",
              "type": "button",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                Homeworld
                : 
                Kashyyk
</li>,
              <li>
                Homeworld Population
                : 
                1000010
</li>,
              <li>
                Species
                : 
                Wookie
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Homeworld",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homeworld",
                  ": ",
                  "Kashyyk",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Homeworld",
                ": ",
                "Kashyyk",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "Homeworld Population",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homeworld Population",
                  ": ",
                  1000010,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Homeworld Population",
                ": ",
                1000010,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "Species",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Species",
                  ": ",
                  "Wookie",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Species",
                ": ",
                "Wookie",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
